File: clase1/ejer2_DictionaryReplacerKata/DictionaryReplacer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace ejer2_DictionaryReplacerKata {
    public class DictionaryReplacer {

        public string Replace(string text, Dictionary <string ,string> dic) {
            string result = "";
            if (dic != null && dic.Count > 0) {
                result = text;
                foreach(var key in dic.Keys) {
                    result = result.Replace("$" + key + "$", dic[key]);
                }
            }

            return result;
        }
    }
}

File: javascript/DictionaryReplacer/DictionaryReplacer.js
function DictionaryReplacer(){
	this.replace = (text, dic)=>{
		let result = '';

		if (dic && Object.keys(dic).length){
			result = text;
			for(let key in dic){
				result = result.replace('$' + key + '$', dic[key]); 
			}	
		}
		
		return result;
	};
}